More C++11 fixes

vadz opened the pull request:
Address a few more items of #22684, notably fixes all "TODO-C++11" comments in the code.

vadz committed:
Replace StateEventProcessor used in wxWebRequest code with lambda

No real changes, just simplify the code now that we can use C++11.

This commit is best viewed with Git --color-moved option.

vadz committed:
Make class used as template parameter local now that we can

Address a minor TODO-C++11 comment in the dialogs sample by moving the
class declaration into the function using it, now that we don't have to
keep it outside it.

This commit is best viewed with Git --color-moved option.

vadz committed:
Use variadic templates in custom wxFileDialog controls code

This gets rid of a hack with unnecessary TextCtrlImpl ctor argument and
allows to reuse AddToLayoutAndReturn() for ChoiceImpl, requiring 2
arguments and not (at most) 1, as all the other classes, too.

No real changes, this is just a simplification/prettification.

vadz committed:
Use std::modf() instead of computing fractional part manually

No real changes, just a simplification.

vadz committed:
Use lambdas with wxImage::ApplyToAllPixels()

This makes the code simpler, as it avoids the need for packing/unpacking
the operation parameters into helper classes and allows to remove
WeightValue completely, and potentially more efficient, as the calls
should be inlined now instead of being performed via a function pointer.

No real changes otherwise.

vadz committed:
Define __WXFUNCTION__ as __func__ and don't use it any more

Use __func__ without checking if the compiler supports it, it's part of
C++11 and so should be supported by all compilers.

Also use __func__ instead of __WXFUNCTION__ in our own code.

No real changes.

PBfordev commented on samples/dialogs/dialogs.cpp:
```diff
@@ -3117,6 +3089,27 @@ TestDefaultActionDialog::TestDefaultActionDialog( wxWindow *parent ) :
+        }
+
+    private:
+        void OnEnter(wxCommandEvent& WXUNUSED(event))
```
I wonder if it was not better in new code to use lambdas for one-line event handlers like this...

vadz replied:
Good point, thanks! Done now.

vadz committed:
Remove EnterHandlingTextCtrl class from the dialogs sample

This code was simplified by 16473d9b19 (Make class used as template
parameter local now that we can, 2022-11-11) but we can simplify it even
more by just getting rid of this class completely and using a small
lambda as the event handler.

No real changes.
